src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas2/StatusUpdateService.kt

```diff
@@ -48,6 +54,16 @@ class StatusUpdateService(
         ValidatableActionResult.GeneralValidationError("The status ${statusUpdate.newStatus} is not valid"),
       )
 
+    var statusDetails: List<Cas2PersistedApplicationStatusDetail>? = null
+    if (!statusUpdate.newStatusDetails.isNullOrEmpty()) {
+      statusDetails = statusUpdate.newStatusDetails.map { detail ->
+        findStatusDetailOnStatus(status, detail)
+          ?: return AuthorisableActionResult.Success(
+            ValidatableActionResult.GeneralValidationError("The status detail $detail is not valid"),
+          )
+      }
+    }
```

Comment:
instead of having a mutable list 'var statusDetails' how about doing something like the following code. Then statusDetails will be non nullable removing need for ? when accessing it for the forEach

```
val statusDetails = if (statusUpdate.newStatusDetails.isNullOrEmpty()) {
  emptyList()
} else {
  statusUpdate.newStatusDetails.map { detail ->
    findStatusDetailOnStatus(status, detail)
      ?: return AuthorisableActionResult.Success(
        ValidatableActionResult.GeneralValidationError("The status detail $detail is not valid"),
      )
  }
}
```

Reply:
that's some nice Kotlin-y syntax, will use!